### PR TITLE
Update to Go 1.14.3 with testing using 1.13.11

### PR DESCRIPTION
**Pull request by danielnelson**
Linux/Official packages: 1.14.3 + 1.13.11
Mac: 1.13.? (unchanged)
Windows: 1.14.2

### Required for all PRs:

- [ ] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 version: "{build}"
 
+image: Visual Studio 2019
```
Comment (ssoroka) [inline appveyor.yml:3]: ?

Comment (danielnelson) [reply appveyor.yml:3]: This gives a different base image that has Go 1.14.2 preinstalled in C:\go, it appears there is always a version of Go installed so if we install it ourselves we need to put it in another location and adjust the PATH and perhaps some other Go environment variables.

https://www.appveyor.com/docs/windows-images-software/#golang

File: internal/tls/common.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ var tlsVersionMap = map[string]uint16{
 	"TLS10": tls.VersionTLS10,
 	"TLS11": tls.VersionTLS11,
 	"TLS12": tls.VersionTLS12,
+	"TLS13": tls.VersionTLS13,
```
Comment (reimda) [inline internal/tls/common.go:9]: Are these here so people can manually use TLS13 or the new ciphers?  Are we not defaulting to the newest yet because they're not available in all go versions we target?

File: plugins/inputs/clickhouse/clickhouse.go [hunk 3]
```diff
@@ -205,6 +205,10 @@ func (ch *ClickHouse) Gather(acc telegraf.Accumulator) (err error) {
 	return nil
 }
 
+func (ch *ClickHouse) Stop() {
+	ch.client.CloseIdleConnections()
```
Comment (reimda) [inline plugins/inputs/clickhouse/clickhouse.go:209]: This looks unrelated to the rest of the PR.

File: internal/tls/common.go (hunk 2 above)
Comment (ssoroka) [reply internal/tls/common.go:9]: fyi looks like go 1.12 had support for tls 1.3

File: plugins/inputs/clickhouse/clickhouse.go (hunk 3 above)
Comment (danielnelson) [reply plugins/inputs/clickhouse/clickhouse.go:209]: I removed the Go 1.11 support code and this is moved over from `clickhouse_go1.12.go`.  I could split thee pull request?

Comment (reimda) [reply plugins/inputs/clickhouse/clickhouse.go:209]: Sorry, I missed the deleted clickhouse files.  It makes sense that this was moved from clichouse_go1.12.go.  No need to split the pr.

File: internal/tls/common.go (hunk 2 above)
Comment (danielnelson) [reply internal/tls/common.go:9]: These are moved over from `common_go112.go`, it is possible to set some advanced TLS options in additiona to the options that are shown in the sample config: https://github.com/influxdata/telegraf/blob/master/docs/TLS.md.  Actually looks like I need to update these docs.

**Comment by danielnelson** [comment]
Updated TLS documentation, removing references to Go 1.12 and older.  Also corrected documentation to indicate that the advanced settings are for the **server** config only.
